fbmessenger/__init__.py

```diff
@@ -216,25 +221,41 @@ def upload_attachment(self, attachment, timeout=None):
             raise ValueError('Attachment may not have `quick_replies`')
         r = self.session.post(
             '{graph_url}/me/message_attachments'.format(graph_url=self.graph_url),
-            params={
-                'access_token': self.page_access_token
-            },
+            params=self.auth_args,
             json={
                 'message':  attachment.to_dict()
             },
             timeout=timeout
         )
         return r.json()
 
+    @staticmethod
+    def generate_appsecret_proof(access_token, app_secret):
```

Comment:
Maybe we should remove staticmethod, access_token and app_secret.
And use self.access_token and self.app_secret